tests/components/plex/test_media_players.py

```diff
@@ -1,41 +1,33 @@
 """Tests for Plex media_players."""
 from plexapi.exceptions import NotFound
 
-from homeassistant.components.plex.const import DOMAIN, SERVERS
+from .payloads import PLAYER_PLEXWEB_RESOURCES
 
 from tests.async_mock import patch
 
 
-async def test_plex_tv_clients(hass, entry, mock_plex_account, setup_plex_server):
+async def test_plex_tv_clients(hass, entry, setup_plex_server, requests_mock):
     """Test getting Plex clients from plex.tv."""
-    resource = next(
-        x
-        for x in mock_plex_account.resources()
-        if x.name.startswith("plex.tv Resource Player")
-    )
-    with patch.object(resource, "connect", side_effect=NotFound):
-        mock_plex_server = await setup_plex_server()
+
+    with patch("plexapi.myplex.MyPlexResource.connect", side_effect=NotFound):
+        await setup_plex_server()
         await hass.async_block_till_done()
 
-    server_id = mock_plex_server.machineIdentifier
-    plex_server = hass.data[DOMAIN][SERVERS][server_id]
     media_players_before = len(hass.states.async_entity_ids("media_player"))
 
     # Ensure one more client is discovered
+    requests_mock.get("/resources", text=PLAYER_PLEXWEB_RESOURCES)
```

Comment:
What ensures that another client is discovered in this test? It reads like it just unloads the config then re-runs it and another client shows up.  Was /resources returning an empty response before?

Reply:
During the first attempt we force a connection failure to the resource (Plex client) by raising a `plexapi.exceptions.NotFound` exception in response to the `connect` call. As we don't patch that method the second time around, it will succeed and an additional player will be available.

Reply:
Got it.  Seems pretty clear now, so not sure why I couldn't see it.

This order might make it slighly more clear:

```
    requests_mock.get("/resources", text=PLAYER_PLEXWEB_RESOURCES)
    with patch("plexapi.myplex.MyPlexResource.connect", side_effect=NotFound):
        await setup_plex_server()
        await hass.async_block_till_done()

    media_players_before = len(hass.states.async_entity_ids("media_player"))
    await hass.config_entries.async_unload(entry.entry_id)

    # Ensure one more client is discovered
    await setup_plex_server()
    await hass.async_block_till_done()

```